src/ray/core_worker/core_worker.cc

```diff
@@ -1441,56 +1460,102 @@ Status CoreWorker::SealExisting(const ObjectID &object_id,
 
 Status CoreWorker::ExperimentalChannelReadRelease(
     const std::vector<ObjectID> &object_ids) {
-  RAY_CHECK(object_ids.size() == 1);
-  return experimental_mutable_object_manager_->ReadRelease(object_ids[0]);
+  RAY_CHECK_EQ(object_ids.size(), 1UL);
+  return experimental_mutable_object_provider_->ReadRelease(object_ids[0]);
 }
 
-Status CoreWorker::ExperimentalChannelRegisterReader(const ObjectID &object_id) {
-  return ExperimentalChannelRegisterWriterOrReader(object_id, /*is_writer=*/false);
+Status CoreWorker::ExperimentalRegisterMutableObjectWriter(const ObjectID &object_id,
+                                                           const NodeID *node_id) {
+  experimental_mutable_object_provider_->RegisterWriterChannel(object_id, node_id);
+  return Status::OK();
 }
 
-Status CoreWorker::ExperimentalChannelRegisterWriter(const ObjectID &object_id) {
-  return ExperimentalChannelRegisterWriterOrReader(object_id, /*is_writer=*/true);
-}
+Status CoreWorker::ExperimentalRegisterMutableObjectReaderRemote(
+    const ObjectID &object_id,
+    const NodeID &reader_node,
+    const std::vector<std::string> &worker_ids,
+    int buffer_size_bytes,
+    int64_t num_readers,
+    ObjectID &reader_ref) {
+  std::ofstream f;
+  f.open("/tmp/blah", std::ofstream::app);
+  f << "ExperimentalRegisterMutableObjectReaderRemote, worker id is "
+    << worker_context_.GetWorkerID() << std::endl;
+
+  for (const std::string &worker_id_hex : worker_ids) {
+    WorkerID worker_id = WorkerID::FromHex(worker_id_hex);
+    rpc::Address addr;
+    {
+      std::promise<void> promise;
+      RAY_CHECK(
+          gcs_client_->Workers()
```

Comment:
Hmm I realized if we pass a list of the reader WorkerIDs to the Channel constructor instead of a node ID, this can be done more cleanly. Plus it will also work better for cases where the readers are on different hood, i.e. this code can take care of registering with each different reader node instead of having to create different Python Channels. The current method is good for now, though, so can you put this in as a TODO?